src/snowflake/sqlalchemy/custom_commands.py

```diff
@@ -42,24 +42,25 @@ def __init__(self, command):
         def __repr__(self):
             if self.command == "INSERT":
                 sets, sets_tos = zip(*self.set.items())
-                return "WHEN NOT MATCHED{} THEN {} ({}) VALUES ({})".format(
-                    " AND %s" % self.predicate if self.predicate is not None else "",
-                    self.command,
-                    ", ".join(sets),
-                    ", ".join(map(str, sets_tos)),
-                )
+                return f"""
+WHEN NOT MATCHED\
+{f" AND {str(self.predicate if self.predicate is not None else '')}"} \
```

Comment:
Isn't `str` redundant here? I always thought that Python will call `str` on f-string pieces because it's embedding it in a string.

```pycon
>>> class A:
...     def __init__(self, val=None):
...             self.val = val
...     def __str__(self):
...             print("str")
...             return str(self.val)
...
>>> a=A('a')
>>> print(f"{a}")
str
a
>>> b=A('b')
>>> print(f"{b}")
str
b
```

Reply:
I called `str` because "%s" was used in the original code. I did some more search and it's another rabbit hole: https://peps.python.org/pep-0498/#format-specifiers

but in short, the call priority chain is `__format__` -> `__str__` -> `__repr__` when using f-strings in my test.
so I think `str` is needed here just being cautious.

```python
class A:
    def __str__(self):
        return 'str called'

    def __repr__(self):
        return 'repr called'

    def __format__(self, format_spec):
        return 'format called'


a = A()
str_a = f"{a}"
print(str_a)  # 'format called'
```